### PR TITLE
Add CodeQL security scanning

### PR DESCRIPTION
**Pull request by jhutchings1**
Hi, I'm a PM on the GitHub security team. This repository is eligible to try the new [GitHub Advanced Security code scanning](https://help.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/about-code-scanning) beta.

Code scanning runs a static analysis tool called CodeQL which scans your code at build time to find any potential security issues. We've tuned the set of queries to be only the most severe, most precise issues. We'll show alerts in the security tab, and we'll show alerts for any net new vulnerabilities on pull requests as well. We've tried to make this super developer friendly, but we'd love your feedback as we work through the beta. 

If you're interested in trying it out, you can merge this pull request to set up the Actions workflow.

### CODE REVIEW

**Comment by jhutchings1** [comment]
I confirmed that the CodeQL build works, and found no known security issues. For this project, the build took less than 4 minutes. Hope you're up for merging this!

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: "Code scanning - action"
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 19 * * 0'
```
Comment (faf0-addepar) [inline .github/workflows/codeql.yml:7]:
```suggestion
    # Every Monday at 1pm
    - cron: '0 13 * * 1'
```

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
```
Comment (faf0-addepar) [inline .github/workflows/codeql.yml:46]:
```suggestion
```

to clean it up a bit

**Comment by jhutchings1** [comment]
Thanks @faf0-addepar! I've incorporated your suggestions.

**Pull request closed without merging**